### ISSUE TITLE
Upgrade to Apache POI 4.0

### ISSUE THREAD

**Issue by mjul** [mjul/docjure#78]
The Apache POI project used by Docjure has released version 4.0.0.
Docjure should be upgraded from the 3.x series to use the latest version. 
Read about Apache POI here: [http://poi.apache.org](http://poi.apache.org)

**Comment by manuelherzog** [comment]
Hey,

right now i'm doing the migration, as i need the newer version for other POI tasks and libs.

I'm getting to multiple problems before i could start, lein test did not run because the Classloader was not found. I updated pomegranate and then get the error that there is no writeable Classloader anymore. Then i removed that one test that was failing because it can't add the test-data to the classpath.

I upgraded to POI 4.0.1 and it works quite easily.

But, now i get the exception "Truncated ZIP file" on one test only. Rolled back and noticed that the stream that was written to was not closed properly.

Sending you the pull request with 3 commits, if you want to fix the two problems not directly attached to the migration in another way

Best
Manuel
